src/execution_tree/primitives/column_set.cpp

```diff
@@ -0,0 +1,360 @@
+                }
+
+                if (value_dimnum == 2)
+                {
+                    HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                        "phylanx::execution_tree::primitives::"
+                        "column_set_operation::column_set1d",
+                        generate_error_message(
+                            "can not store matrix in a vetor", name_,
+                            codename_));
+                }
+
+                auto init_list = create_list_col_set(
+                    col_start, col_stop, step, args[0].size());
+
+                auto sv = blaze::elements(args[0].vector(), init_list);
+
+                if (value_dimnum == 0)
+                {
+                    blaze::DynamicVector<double> temp(
+                        sv.size(), args[4].scalar());
+                    sv = temp;
+                    return primitive_argument_type{ir::node_data<double>(0)};
+                }
+
+                auto temp = args[4].vector();
+
+                if (sv.size() != temp.size())
+                {
+                    HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                        "phylanx::execution_tree::primitives::"
+                        "column_set_operation::column_set1d",
+                        generate_error_message("size mismatch, please check "
+                                               "your parameters or set vector",
+                            name_, codename_));
+                }
+
+                sv = temp;
+                return primitive_argument_type{ir::node_data<double>(0)};
+            }
+
+            primitive_argument_type column_set2d(args_type&& args) const
+            {
+                auto col_start = args[1].scalar();
+                auto col_stop = args[2].scalar();
+                auto step_col = args[3].scalar();
+                auto row_start = 0;
+                auto row_stop = args[0].dimensions()[0];
+                auto step_row = 1;
+                auto num_matrix_rows = args[0].dimensions()[0];
+                auto num_matrix_cols = args[0].dimensions()[1];
+                auto value_dimnum = args[4].num_dimensions();
+
+                if (step_col == 0)
+                {
+                    HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                        "phylanx::execution_tree::primitives::"
+                        "column_set_operation::set2d",
+                        generate_error_message(
+                            "step can not be zero", name_, codename_));
+                }
+
+                auto init_list_col = create_list_col_set(
+                    col_start, col_stop, step_col, num_matrix_cols);
+
+                auto sm = blaze::columns(args[0].matrix(), init_list_col);
+
+                if (value_dimnum == 0)
+                {
+                    blaze::DynamicMatrix<double> data(
+                        sm.rows(), sm.columns(), args[4].scalar());
+                    sm = data;
```

Comment:
Can this use `std::move`? Also, more of this below...

Reply:
As per our discussion, we decided to not use std::move in this case and leave this as is.